src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -832,32 +772,110 @@ else if (JAVADOC_TOKENS.equals(propertyName)) {
      * @return the since version of the property.
      * @throws MacroExecutionException if the since version could not be extracted.
      */
-    public static String getSinceVersion(String moduleName, DetailNode moduleJavadoc,
-                                         String propertyName, DetailNode propertyJavadoc)
+    public static String getSinceVersion(String moduleName,
+                                         DetailNode moduleJavadoc,
+                                         String propertyName,
+                                         DetailNode propertyJavadoc)
             throws MacroExecutionException {
-        final String sinceVersion;
-        final String superClassSinceVersion = SINCE_VERSION_FOR_INHERITED_PROPERTY
-                   .get(moduleName + DOT + propertyName);
-        if (superClassSinceVersion != null) {
-            sinceVersion = superClassSinceVersion;
+        final String key = moduleName + DOT + propertyName;
+        final String fallbackSince = SINCE_VERSION_FOR_INHERITED_PROPERTY.get(key);
+
+        final String moduleSince = getSinceVersionFromJavadoc(moduleJavadoc);
```

Comment:
separated concern could live in dedicated place (function)

is impl. detail of concern below.